web/html/src/manager/systems/virtual-list.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import Network from "utils/network";
+import * as Systems from "components/systems";
+import { Utils } from "utils/functions";
+import { Table } from "components/table/Table";
+import { Column } from "components/table/Column";
+import SpaRenderer from "core/spa/spa-renderer";
+import { IconTag } from "components/icontag";
+import { SearchField } from "components/table/SearchField";
+
+// See java/code/src/com/suse/manager/webui/templates/systems/virtual-list.jade
+type Props = {
+  /** Locale of the help links */
+  docsLocale: string;
+  isAdmin: boolean;
+};
+
+function VirtualSystems(props: Props) {
+  const [selectedSystems, setSelectedSystems] = useState([]);
+  const [selectedSystemsCount, setCount] = useState(0);
+
+  const handleSelectedSystems = (data) => {
+    setSelectedSystems(data);
+  };
+
+  useEffect(()=> {
+    setCount(selectedSystems.length);
+    document.getElementById("header_selcount").innerHTML =
+      "<span id='spacewalk-set-system_list-counter' class='badge'>" +
+      selectedSystemsCount.toString() +
+      "</span>" +
+      (selectedSystemsCount === 1 ? "system selected" : "systems selected");
+  }, [handleSelectedSystems]);
+
+  const addToSSM = () => {
+    var url = "/rhn/manager/systems/addToSsm";
+    Network.post(url, selectedSystems);
+  };
+
+  const searchData = (datum, criteria) => {
+    if (criteria) {
+      return datum.name.toLowerCase().includes(criteria.toLowerCase());
+    }
+    return true;
+  };
+
+  return (
+    <>
+      <h1>
+        <IconTag type="header-system" />
+        {t("Virtual Systems")}
+        <a
+          href={`/docs/${props.docsLocale}/reference/systems/systems-list.html`}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <IconTag type="header-help" />
+        </a>
+      </h1>
+
+      <div>
+        <button className="btn btn-default" onClick={addToSSM}>
+          {t("Add Selected to SSM")}
+        </button>
+      </div>
+
+      <Table
+        data="/rhn/manager/api/systems/list/virtual"
+        identifier={(item) => item.uuid}
+        initialSortColumnKey="hostServerName"
+        selectable
+        isSelectEnabled={(item) => item.hasOwnProperty("virtualSystemId")}
+        selectedItems={selectedSystems}
+        onSelect={handleSelectedSystems}
+        initialItemsPerPage={window.userPrefPageSize}
+        searchField={<SearchField filter={searchData} placeholder={t("Filter by System name: ")} />}
```

Comment:
Small inconsistency, in other views placeholder strings don't have trailing symbols (`: `), search for `placeholder={t("Filter by` for a few examples.

Reply:
👍🏻